### PR TITLE
Input Lock

### PR DESCRIPTION
**Pull request by vLesio**
# Description
Input Lock was implemented to deal with problem with InputActions, that can be accessed by multiple sources on any given time. It could lead to denying of `Enable` or `Disable` functions by other scripts than the one performing the action.

Lock enables Input Action only if no objects want this action to be disabled.

From now on, instead of using `CInput.InputAction.Teleport.Disable()`, use `CInput.Teleport.Lock()`. The same goes for `Movement`, `Interactions` etc.

### CODE REVIEW

**Comment by empieniazek** [comment]
> Possible implications
> Worse cross-platform compatibility, problems with building for different platforms might arise

Do I understand correctly, that this is risk we have to take?
Is there any other way to implement the functionality without the risk of building problems?

File: Assets/Scripts/CustomInput/ILockable.cs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+﻿namespace CustomInput {
+    public interface ILockable {
```
Comment (empieniazek) [inline Assets/Scripts/CustomInput/ILockable.cs:2]: I can't find any usage of ILockable interface in files in this PR. Can you tell me where do we use it?

**Comment by vLesio** [comment]
> Do I understand correctly, that this is risk we have to take?
Is there any other way to implement the functionality without the risk of building problems?

I changed implementation to not include functionalities provided with .NET Framework and rolled back to previous settings. This comes at significant cost of probable code maintenance later, but hopefully we won't have to do that.

**Comment by vLesio** [comment]
We could also leverage .NET Framework and have some tests later on compatibilty with platforms we want the game published on. Not sure tho

**Review by empieniazek — APPROVED** [review]

**Pull request merged**